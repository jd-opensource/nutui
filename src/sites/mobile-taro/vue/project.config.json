{
  "miniprogramRoot": "./dist",
  "projectname": "@nutui/nutui-taro-mobile",
  "description": "nutui-taro-vue",
  "appid": "touristappid",
  "setting": {
    "urlCheck": true,
    "es6": false,
    "postcss": false,
    "minified": false
  },
  "compileType": "miniprogram"
}
