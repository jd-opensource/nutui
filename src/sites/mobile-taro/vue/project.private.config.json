{
  "setting": {},
  "condition": {
    "plugin": {
      "list": []
    },
    "game": {
      "list": []
    },
    "gamePlugin": {
      "list": []
    },
    "miniprogram": {
      "list": [
        {
          "name": "business/pages/address/index",
          "pathName": "business/pages/address/index",
          "query": "",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/noticebar/index",
          "pathName": "feedback/pages/noticebar/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/progress/index",
          "pathName": "feedback/pages/progress/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/swiper/index",
          "pathName": "feedback/pages/swiper/index",
          "query": "",
          "scene": null
        }
      ]
    }
  }
}