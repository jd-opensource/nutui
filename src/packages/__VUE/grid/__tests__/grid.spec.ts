import { h, nextTick } from 'vue';
import { config, mount } from '@vue/test-utils';
import { vi } from 'vitest';
import { useRouter } from 'vue-router';
import Grid from '../index.vue';
import GridItem from '../../griditem/index.vue';

afterAll(() => {
  config.global.components = {};
});

// mock module
vi.mock('vue-router', () => ({
  useRouter: vi.fn()
}));

test('should render square correctly', () => {
  const wrapper = mount(Grid, {
    props: {
      square: true,
      columnNum: 2
    },
    slots: {
      default: [GridItem, GridItem, GridItem, GridItem]
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render gutter correctly', () => {
  const wrapper = mount(Grid, {
    props: {
      gutter: 20
    },
    slots: {
      default: [GridItem, GridItem, GridItem, GridItem, GridItem, GridItem, GridItem, GridItem]
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should render default slot correctly', () => {
  const wrapper = mount(Grid, {
    slots: {
      default: h(GridItem, null, {
        default: () => 'Default Slot'
      })
    }
  });

  expect(wrapper.find('.nut-grid-item__content').html()).toContain('Default Slot');
  expect(wrapper.html()).toMatchSnapshot();
});

test('should emit click correctly', async () => {
  const wrapper = mount(Grid, {
    slots: {
      default: [GridItem]
    }
  });

  wrapper.find('.nut-grid-item').trigger('click');
  await nextTick();

  expect(wrapper.emitted('click')).toHaveLength(1);
});

test('should navifation correctly', async () => {
  // 当 `useRouter()` 时返回 `push` 方法
  const push = vi.fn((url: string) => url);
  (useRouter as any).mockImplementationOnce(() => ({
    push
  }));

  const wrapper = mount(Grid, {
    slots: {
      default: h(GridItem, {
        to: 'http://m.jd.com'
      })
    }
  });
  expect(wrapper.html()).toMatchSnapshot();
  // wrapper.find('.nut-grid-item').trigger('click');
  // await nextTick();
  // expect(push.mock.calls[0][0]).toEqual('/home');
});
