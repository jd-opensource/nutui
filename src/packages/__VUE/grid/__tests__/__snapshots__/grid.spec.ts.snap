// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should navifation correctly 1`] = `
"<view class=\\"nut-grid nut-grid--border\\">
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--clickable\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
</view>"
`;

exports[`should render default slot correctly 1`] = `
"<view class=\\"nut-grid nut-grid--border\\">
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center\\">Default Slot<view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
</view>"
`;

exports[`should render gutter correctly 1`] = `
"<view class=\\"nut-grid\\" style=\\"padding-left: 20px;\\">
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px; margin-top: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px; margin-top: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px; margin-top: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 25%; padding-right: 20px; margin-top: 20px;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--surround nut-grid-item__content--center\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
</view>"
`;

exports[`should render square correctly 1`] = `
"<view class=\\"nut-grid nut-grid--border\\">
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 50%; padding-top: 50%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--square\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 50%; padding-top: 50%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--square\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 50%; padding-top: 50%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--square\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
  <view class=\\"nut-grid-item\\" style=\\"flex-basis: 50%; padding-top: 50%;\\">
    <view class=\\"nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--square\\">
      <view class=\\"nut-grid-item__text\\"></view>
    </view>
  </view>
</view>"
`;
